### ISSUE TITLE
Torsion and Nonbonded NaN

### ISSUE THREAD

**Issue by adw62** [openmm/openmm#2959]
Hello,

I'm attempting to perform alchemical protein-ligand binding free energy calculations for a set of proteins and ligands. I'm finding that a small subset of the ligands I would like to study seem to have some problem with their initial structures.

This problem is manifesting its self as a NaN of the system which occurs immediately during minimization. By removing forces I think I have found that the NaN is occurring in the nonbonded and periodic torsion forces together.

I have tried 'zeroing' the atoms and torsion to see if I could turn them back on one at a time and identify the problem. However, confusingly even after setting the charge, epsilon and k parameters to zero the NaN still occurs.

I have attached my input and the script I'm using for testing here: [ligand_nan.zip](https://github.com/openmm/openmm/files/5730874/ligand_nan.zip). The output I get is:
```
###Defualt simulation
simulation energy before min: nan kJ/mol
Minimizing...
Particle coordinate is nan
###simulation with torsion and Nonbonded removed
removing: NonbondedForce
removing: PeriodicTorsionForce
simulation energy before min: 1391.1564915406307 kJ/mol
Minimizing...
simulation energy after min: 1363.6171470494262 kJ/mol
True
###simulation with torsion and Nonbonded zeroed
simulation energy before min: nan kJ/mol
Minimizing...
Particle coordinate is nan
```

This system does have overlapping atoms as I intend to use it in a dual topology alchemical calculations. I think the minimizer should still be deal with this system and I have tested my script with other overlapped ligands and I can minimize the system without issue but this could be a bad assumption by me.

I would appreciate any help with this problem, particularly if anyone knows why the 'zeroed' forces still NaN.

Thanks,
Alex

**Comment by peastman** [comment]
Sorry for the delay.  I'm just catching up on things after the holidays.

The problem does come from having atoms at the same position.  Even if you set their force parameters to zero that doesn't help, because then the interaction energy is 0/0=nan.

I don't think you really want those atoms to interact with each other, do you?  If they correspond to two different versions of the ligand that you're interpolating between, then I think you want each version to interact with the environment, but you never want them to interact with each other.  Is that right?  If so, you can call `addException()` on the NonbondedForce to tell it they shouldn't interact.

**Comment by adw62** [comment]
No problem, thanks for the reply. 

Okay I see, it is expected that the nan will remain after zeroing the parameters.

You are correct I do not want the two versions of the ligand to see each other. Normally I add exceptions using OpenMMTools however this complicates the example script and so I have omitted it here. However, the nan persists even with the exceptions added by OpenMMTools.

I will write a example script that includes these exceptions explicitly and post back with my result.

**Comment by adw62** [comment]
I have tried adding the exceptions between all pairs of the alchemical atoms and this has resolved the nan in the Nonbonded as you suggested. However, the system still nans overall and the cause looks to be just the torsions now.

For my method I normally remove the torsions, angles and bonds between the two ligand and I have included the code to do this in my new example, attached: [lig_nan_excp.zip](https://github.com/openmm/openmm/files/5771331/lig_nan_excp.zip)

To remove these torsions and angles between the alchemical regions I'm, again, zeroing the parameters. Is the problem, as in the nonbonded case, that there is a nan in the torsion and I can't remove it by zeroing the parameters?
